pkg/system/phase2_creating.go

```diff
@@ -341,7 +343,14 @@ func (r *Reconciler) SetDesiredCoreApp() error {
 						}
 					}
 				}
-						
+			}
+
+			// adding missing Env varibale from default container
+			for j := range r.DefaultCoreApp.Env {
+				if (!existingEnvs[r.DefaultCoreApp.Env[j].Name] && 
+					(r.DefaultCoreApp.Env[j].Value != "" || r.DefaultCoreApp.Env[j].ValueFrom != nil)) {
+					c.Env = append(c.Env, r.DefaultCoreApp.Env[j])
+				}
```

Comment:
I don't think that copy is enough as sometimes we leave envs values empty on purpose so the code will reconcile them with custom logic.  If I will just delete n env that has an empty value (in the template)  it will not get its reconciled value back.

I think a better approach is to take the list from the template then merge into it the current envs then run the reconcile code (the for with the switch statement) on the merged list

Reply:
@nb-ohad is right. just add a `mergeEnvs` function and call it before the envs `for`

Reply:
Just to clarify, you should merge only the missing envs from the  template  into the CR, then the reconcile env loop can stay the same